### PR TITLE
Games show page

### PR DESCRIPTION
**Pull request by altitudor**
Games Show Page created. 
Tests for Game Show Page written. 
Made small updates to the games table. 
All indentation standardized (tab: 2)

### CODE REVIEW

File: app/javascript/react/components/GamesShowComponent.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const title = props.game.title
+  const image = props.game.image
+  const description = props.game.description
+  const creator = props.game.creator
+  const platform = props.game.platform
+  const genre = props.game.genre
+  const site = props.game.site
+  const release_date = props.game.release_date
+  const created_at = props.game.created_at
+
```
Comment (nwalberts) [inline app/javascript/react/components/GamesShowComponent.js:13]: this is too verbose. either pull from props in your JSX, or use object destructuring: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment (down some ways)

e.g. const GamesShowComponent = ( {game} ) => {
    const { title, image, description, creator, platform, genre, site, release_date, created_at } = game

File: app/javascript/react/components/GamesShowComponent.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <>
+      <h2> Hello from GamesShowComponent </h2>
```
Comment (nwalberts) [inline app/javascript/react/components/GamesShowComponent.js:16]: no Dummy data. This should be a formal title on the page or be removed

File: app/javascript/react/components/GamesShowComponent.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+      <div><h3>{title}</h3></div>
+      <div><img src={image}></img></div>
+      <div><h3>{description}</h3></div>
+      <div><h3>{creator}</h3></div>
+      <div><h3>{platform}</h3></div>
+      <div><h3>{genre}</h3></div>
+      <div><h3>{site}</h3></div>
+      <div><h3>{release_date}</h3></div>
+      <div><h3>{created_at}</h3></div>
```
Comment (nwalberts) [inline app/javascript/react/components/GamesShowComponent.js:26]: ...all of these divs are largely just taking up space. what purpose do they serve currently? remove them or use styling if that is your aim 

additionally, we shouldnt have so many h3s. Make this actually look good with CSS and more variation in HTML tags

File: app/javascript/react/containers/GamesIndexContainer.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React, { useState, useEffect } from "react"
+
+const GamesIndexContainer = props => {
+  return(
+    <div> Hello from GamesIndexContainer </div>
+  )
+}
+
+export default GamesIndexContainer
```
Comment (nwalberts) [inline app/javascript/react/containers/GamesIndexContainer.js:9]: all code must be deliverable and ready for users to see. Additionally, this component is out of scope. This is a show page PR, not an index page PR. remove

File: app/javascript/react/containers/GamesShowContainer.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect, Fragment } from "react"
+import { checkPropTypes } from "prop-types"
```
Comment (nwalberts) [inline app/javascript/react/containers/GamesShowContainer.js:2]: please do not use prop-types. they are great, but you would have had to be using them across the project, which we do not want you to do

File: app/javascript/react/containers/GamesShowContainer.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        let error = new Error(errorMessage)
+        throw(error)
+      }
+    })
+    .then(response => response.json())
+    .then(parsedJSON => setGame(parsedJSON))
+    .catch(error => console.error(`Error in fetch: ${error.message}`))
+  }, [])
+
+
```
Comment (nwalberts) [inline app/javascript/react/containers/GamesShowContainer.js:38]: remove extra space

File: app/javascript/react/containers/GamesShowContainer.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  return(
+    <>
+      <GamesShowComponent game = {game} />
+    </>
+  )
```
Comment (nwalberts) [inline app/javascript/react/containers/GamesShowContainer.js:43]: you do not need a fragment here. just return line 41 alone

File: app/models/game.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class Game < ApplicationRecord
+  validates :title, presence: true
+
+end
```
Comment (nwalberts) [inline app/models/game.rb:4]: remove extra space

File: spec/controllers/api/v1/games_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    it "returns all games in the database" do
+      get :index
+      returned_json = JSON.parse(response.body)
+
+      expect(returned_json.length).to eq 3
+      expect(returned_json[0]["title"]).to eq game1.title
+      expect(returned_json[1]["title"]).to eq game2.title
+      expect(returned_json[2]["title"]).to eq game3.title
+    end
+  end
```
Comment (nwalberts) [inline spec/controllers/api/v1/games_controller_spec.rb:26]: because there is not a finished index component feature, we will have to delete this index test from the branch for now

File: spec/controllers/api/v1/games_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      expect(response.content_type).to eq ("application/json")
+    end
+
+    it "returns the specific game" do
+      get :show, params: {id: game1.id}
+      returned_json = JSON.parse(response.body)
+
+      expect(returned_json.length).to eq 12
+      expect(returned_json["title"]).to eq game1.title
+    end
```
Comment (nwalberts) [inline spec/controllers/api/v1/games_controller_spec.rb:42]: there are far more keys to check than just title. Make sure to check them all!

File: spec/models/game_spec.rb [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+
+describe Game do
+  let(:title) { "Valid Title" }
+  let(:image) { "Valid image text" }
+  let(:game) do
+    Game.new({
+      title: title,
+      image: image
+    })
+  end
```
Comment (nwalberts) [inline spec/models/game_spec.rb:11]: lines 6 through 11 can probably fit on one line. remove lines 4 and 5, as they can just be written into your :game variable directly

File: spec/models/game_spec.rb [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+        game_with_blank_attributes.save
+
+        games_data = nil
+
+        games_data = Game.all
+        expect(games_data.count).to eq(0)
+      end
+    end
+  end
+end
```
Comment (nwalberts) [inline spec/models/game_spec.rb:160]: I think there is some confusion here. None of these methods were ones that YOU added. Instead, they all come from ActiveRecord. As such, AR will test them in their library, you do not need to test them. Delete this spec :/

File: spec/support/factory_bot.rb [hunk 13]
```diff
@@ -8,4 +8,7 @@
     password_confirmation { 'password' }
   end
 
+  factory :game do
+    sequence(:title) {|n| "game#{n}"}
+  end
```
Comment (nwalberts) [inline spec/support/factory_bot.rb:13]: this factory....definitely does not look complete

File: app/javascript/react/components/App.js [hunk 14]
```diff
@@ -1,7 +1,19 @@
...
+import GamesIndexContainer from "../containers/GamesIndexContainer"
+import GamesShowContainer from "../containers/GamesShowContainer"
+
+const App = (props) => {
+  return (
+    <BrowserRouter>
+      <Switch>
+        <Route exact path="/games" component={GamesIndexContainer}/>
+        <Route exact path="/games/:id" component={GamesShowContainer}/>
+        <Route path="/" component={GamesIndexContainer}/>
```
Comment (nwalberts) [inline app/javascript/react/components/App.js:13]: exact path here

**Comment by Grasscopper** [comment]
Updated show branch and awaiting pending review on changes.

File: app/javascript/react/components/GamesShowComponent.js [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+      <img src={image}></img>
+      <p>Description: {description}</p>
+      <ul>
+        <li>Creator: {creator}</li>
+        <li>Platform: {platform}</li>
+        <li>Genre: {genre}</li>
+        <li>Player Setup: {number_of_players}</li>
+        <li>Site: {site}</li>
+        <li>Release Date: {release_date}</li>
+
```
Comment (safang99) [inline app/javascript/react/components/GamesShowComponent.js:19]: remove the empty line 19
